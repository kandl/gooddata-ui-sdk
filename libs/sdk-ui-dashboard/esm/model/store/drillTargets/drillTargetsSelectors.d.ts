import { DashboardSelector } from "../types.js";
import { ObjRef } from "@gooddata/sdk-model";
import { ObjRefMap } from "../../../_staging/metadata/objRefMap.js";
import { IDrillTargets } from "./drillTargetsTypes.js";
/**
 * Return all widgets drill targets
 * @alpha
 */
export declare const selectDrillTargets: DashboardSelector<ObjRefMap<IDrillTargets>>;
/**
 * Selects drill targets by widget ref.
 *
 * @alpha
 */
export declare const selectDrillTargetsByWidgetRef: (ref: ObjRef) => DashboardSelector<IDrillTargets | undefined>;
