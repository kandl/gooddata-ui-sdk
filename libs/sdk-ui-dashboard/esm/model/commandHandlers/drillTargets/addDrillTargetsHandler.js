import { call, put, select } from "redux-saga/effects";
import { drillTargetsAdded } from "../../events/drillTargets.js";
import { drillTargetsActions } from "../../store/drillTargets/index.js";
import { selectWidgetsMap } from "../../store/layout/layoutSelectors.js";
import { validateExistingInsightWidget } from "../widgets/validation/widgetValidations.js";
import { selectEnableKPIDashboardDrillFromAttribute } from "../../store/config/configSelectors.js";
import { availableDrillTargetsValidation } from "./validation/availableDrillTargetsValidation.js";
import { validateDrills } from "../common/validateDrills.js";
import { selectIsInEditMode } from "../../store/renderMode/renderModeSelectors.js";
import { validateDrillToCustomUrlParams } from "../common/validateDrillToCustomUrlParams.js";
export function* addDrillTargetsHandler(ctx, cmd) {
    const { payload: { availableDrillTargets }, correlationId, } = cmd;
    const widgets = yield select(selectWidgetsMap);
    const enableKPIDashboardDrillFromAttribute = yield select(selectEnableKPIDashboardDrillFromAttribute);
    const insightWidget = validateExistingInsightWidget(widgets, cmd, ctx);
    const drillTarget = availableDrillTargetsValidation(availableDrillTargets, enableKPIDashboardDrillFromAttribute, ctx, cmd);
    const { ref, uri, identifier } = insightWidget;
    yield put(drillTargetsActions.addDrillTargets({
        identifier,
        uri,
        ref,
        availableDrillTargets: drillTarget,
    }));
    // in edit mode, we need to remove invalid drills in case the insight in the widget changes its drill targets
    // and also validate drill to custom URL parameters
    const isInEditMode = yield select(selectIsInEditMode);
    if (isInEditMode) {
        yield call(validateDrills, ctx, cmd, [insightWidget]);
        yield call(validateDrillToCustomUrlParams, [insightWidget]);
    }
    return drillTargetsAdded(ctx, ref, availableDrillTargets, correlationId);
}
//# sourceMappingURL=addDrillTargetsHandler.js.map