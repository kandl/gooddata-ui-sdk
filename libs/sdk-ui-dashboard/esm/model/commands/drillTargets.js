// (C) 2021-2022 GoodData Corporation
/**
 * Create AddDrillTargets {@link AddDrillTargets} command.
 *
 * @param ref - Unique widget ref
 * @param availableDrillTargets - Available widget drill targets {@link @gooddata/sdk-ui#IAvailableDrillTargets}
 * @param correlationId - specify correlation id to use for this command. this will be included in all
 * @returns AddDrillTargets command
 *
 * @alpha
 */
export function addDrillTargets(ref, availableDrillTargets, correlationId) {
    return {
        type: "GDC.DASH/CMD.DRILL_TARGETS.ADD",
        correlationId,
        payload: {
            ref,
            availableDrillTargets,
        },
    };
}
//# sourceMappingURL=drillTargets.js.map