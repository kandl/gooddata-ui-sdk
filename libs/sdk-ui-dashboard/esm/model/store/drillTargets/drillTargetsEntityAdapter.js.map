{"version":3,"file":"drillTargetsEntityAdapter.js","sourceRoot":"","sources":["../../../../src/model/store/drillTargets/drillTargetsEntityAdapter.ts"],"names":[],"mappings":"AAAA,qCAAqC;AACrC,OAAO,EAAE,mBAAmB,EAAE,MAAM,kBAAkB,CAAC;AACvD,OAAO,EAAU,eAAe,EAAE,MAAM,qBAAqB,CAAC;AAE9D,OAAO,EAAE,MAAM,EAAE,MAAM,uBAAuB,CAAC;AAG/C,MAAM,CAAC,MAAM,mBAAmB,GAAG,mBAAmB,CAAgB;IAClE,QAAQ,EAAE,CAAC,YAAY,EAAE,EAAE,CAAC,eAAe,CAAC,qBAAqB,CAAC,YAAY,CAAC,CAAC;CACnF,CAAC,CAAC;AAEH,MAAM,qBAAqB,GAAG,CAAC,OAAsB,EAAU,EAAE;IAC7D,OAAO,MAAM,CAAC,OAAO,CAAC,GAAG,EAAE,OAAO,CAAC,UAAU,CAAC,CAAC;AACnD,CAAC,CAAC"}