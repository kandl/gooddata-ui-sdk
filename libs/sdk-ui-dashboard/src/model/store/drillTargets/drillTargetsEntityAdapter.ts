// (C) 2021-2023 GoodData Corporation
import { createEntityAdapter } from "@reduxjs/toolkit";
import { ObjRef, serializeObjRef } from "@gooddata/sdk-model";

import { objRef } from "../../utils/objRef.js";
import { IDrillTargets } from "./drillTargetsTypes.js";

export const drillTargetsAdapter = createEntityAdapter<IDrillTargets>({
    selectId: (drillTargets) => serializeObjRef(getIdFromDrillTargets(drillTargets)),
});

const getIdFromDrillTargets = (targets: IDrillTargets): ObjRef => {
    return objRef(targets.uri, targets.identifier);
};
