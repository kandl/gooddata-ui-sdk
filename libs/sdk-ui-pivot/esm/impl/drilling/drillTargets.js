// (C) 2007-2022 GoodData Corporation
import { getIntersectionAttributes, } from "@gooddata/sdk-ui";
export function getAvailableDrillTargets(dv, measureGroupDimension, columnHeadersPosition) {
    const measureDescriptors = dv
        .meta()
        .measureDescriptors()
        .map((measure) => ({
        measure,
        attributes: dv.meta().attributeDescriptors(),
    }));
    const dimensionIndex = measureGroupDimension === "rows" && columnHeadersPosition === "left" ? 1 : 0;
    const attributeItems = dv
        .meta()
        .attributeDescriptorsForDim(dimensionIndex)
        .map((attribute, _index, attributes) => ({
        attribute,
        intersectionAttributes: getIntersectionAttributes(attribute, attributes),
    }));
    return {
        measures: measureDescriptors,
        attributes: attributeItems,
    };
}
//# sourceMappingURL=drillTargets.js.map