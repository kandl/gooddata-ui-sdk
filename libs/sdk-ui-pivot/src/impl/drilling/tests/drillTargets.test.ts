// (C) 2021 GoodData Corporation
import {
    TwoMeasuresWithTwoRowAndTwoColumnAttributes,
    OneMeasureAndRepeatingRowAttributesOnDifferentPositions,
} from "./drilling.fixture.js";
import { getAvailableDrillTargets } from "../drillTargets.js";

describe("getAvailableDrillTargets", () => {
    it("gets propper available drill targets - ignoring column attributes", () => {
        expect(getAvailableDrillTargets(TwoMeasuresWithTwoRowAndTwoColumnAttributes)).toMatchSnapshot();
    });

    it("gets propper available drill targets for duplicated row attribute", () => {
        expect(
            getAvailableDrillTargets(OneMeasureAndRepeatingRowAttributesOnDifferentPositions),
        ).toMatchSnapshot();
    });
});
