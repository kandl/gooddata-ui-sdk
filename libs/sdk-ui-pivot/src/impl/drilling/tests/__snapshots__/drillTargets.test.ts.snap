// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAvailableDrillTargets > gets propper available drill targets - ignoring column attributes 1`] = `
{
  "attributes": [
    {
      "attribute": {
        "attributeHeader": {
          "formOf": {
            "identifier": "attr.product.id",
            "name": "Product",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1054",
          },
          "identifier": "label.product.id.name",
          "localIdentifier": "a_label.product.id.name",
          "name": "Product Name",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1055",
        },
      },
      "intersectionAttributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
      ],
    },
    {
      "attribute": {
        "attributeHeader": {
          "formOf": {
            "identifier": "attr.owner.department",
            "name": "Department",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1088",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1088",
          },
          "identifier": "label.owner.department",
          "localIdentifier": "a_label.owner.department",
          "name": "Department",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1089",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1089",
        },
      },
      "intersectionAttributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.department",
              "name": "Department",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1088",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1088",
            },
            "identifier": "label.owner.department",
            "localIdentifier": "a_label.owner.department",
            "name": "Department",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1089",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1089",
          },
        },
      ],
    },
  ],
  "measures": [
    {
      "attributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.department",
              "name": "Department",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1088",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1088",
            },
            "identifier": "label.owner.department",
            "localIdentifier": "a_label.owner.department",
            "name": "Department",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1089",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1089",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.opportunitysnapshot.forecastcategory",
              "name": "Forecast Category",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1062",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1062",
            },
            "identifier": "label.opportunitysnapshot.forecastcategory",
            "localIdentifier": "a_label.opportunitysnapshot.forecastcategory",
            "name": "Forecast Category",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1063",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1063",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.region",
              "name": "Region",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1086",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "identifier": "label.owner.region",
            "localIdentifier": "a_label.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1087",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
        },
      ],
      "measure": {
        "measureHeaderItem": {
          "format": "$#,##0.00",
          "identifier": "aangOxLSeztu",
          "localIdentifier": "m_aangOxLSeztu",
          "name": "Amount",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1271",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1271",
        },
      },
    },
    {
      "attributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.department",
              "name": "Department",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1088",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1088",
            },
            "identifier": "label.owner.department",
            "localIdentifier": "a_label.owner.department",
            "name": "Department",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1089",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1089",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.opportunitysnapshot.forecastcategory",
              "name": "Forecast Category",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1062",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1062",
            },
            "identifier": "label.opportunitysnapshot.forecastcategory",
            "localIdentifier": "a_label.opportunitysnapshot.forecastcategory",
            "name": "Forecast Category",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1063",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1063",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.region",
              "name": "Region",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1086",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "identifier": "label.owner.region",
            "localIdentifier": "a_label.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1087",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
        },
      ],
      "measure": {
        "measureHeaderItem": {
          "format": "$#,##0.00",
          "identifier": "acugFHNJgsBy",
          "localIdentifier": "m_acugFHNJgsBy",
          "name": "Won",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1272",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1272",
        },
      },
    },
  ],
}
`;

exports[`getAvailableDrillTargets > gets propper available drill targets for duplicated row attribute 1`] = `
{
  "attributes": [
    {
      "attribute": {
        "attributeHeader": {
          "formOf": {
            "identifier": "attr.product.id",
            "name": "Product",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1054",
          },
          "identifier": "label.product.id.name",
          "localIdentifier": "a_label.product.id.name",
          "name": "Product Name",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1055",
        },
      },
      "intersectionAttributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
      ],
    },
    {
      "attribute": {
        "attributeHeader": {
          "formOf": {
            "identifier": "attr.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1086",
          },
          "identifier": "label.owner.region",
          "localIdentifier": "a_label.owner.region",
          "name": "Region",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1087",
        },
      },
      "intersectionAttributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.region",
              "name": "Region",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1086",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "identifier": "label.owner.region",
            "localIdentifier": "a_label.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1087",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
        },
      ],
    },
    {
      "attribute": {
        "attributeHeader": {
          "formOf": {
            "identifier": "attr.product.id",
            "name": "Product",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1054",
          },
          "identifier": "label.product.id.name",
          "localIdentifier": "product.name.test",
          "name": "Product Name",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1055",
        },
      },
      "intersectionAttributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.region",
              "name": "Region",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1086",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "identifier": "label.owner.region",
            "localIdentifier": "a_label.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1087",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "product.name.test",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
      ],
    },
  ],
  "measures": [
    {
      "attributes": [
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "a_label.product.id.name",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.owner.region",
              "name": "Region",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1086",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1086",
            },
            "identifier": "label.owner.region",
            "localIdentifier": "a_label.owner.region",
            "name": "Region",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1087",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1087",
          },
        },
        {
          "attributeHeader": {
            "formOf": {
              "identifier": "attr.product.id",
              "name": "Product",
              "ref": {
                "uri": "/gdc/md/referenceworkspace/obj/1054",
              },
              "uri": "/gdc/md/referenceworkspace/obj/1054",
            },
            "identifier": "label.product.id.name",
            "localIdentifier": "product.name.test",
            "name": "Product Name",
            "ref": {
              "uri": "/gdc/md/referenceworkspace/obj/1055",
            },
            "uri": "/gdc/md/referenceworkspace/obj/1055",
          },
        },
      ],
      "measure": {
        "measureHeaderItem": {
          "format": "$#,##0.00",
          "identifier": "aangOxLSeztu",
          "localIdentifier": "m_aangOxLSeztu",
          "name": "Amount",
          "ref": {
            "uri": "/gdc/md/referenceworkspace/obj/1271",
          },
          "uri": "/gdc/md/referenceworkspace/obj/1271",
        },
      },
    },
  ],
}
`;
